### PR TITLE
Created .devcontainer.json file for Github codespaces

### PR DESCRIPTION
**Pull request by Akshit42-hue**
Signed-off-by: Akshit42-hue <patelakshit2025@gmail.com>

Issue #5464

### CODE REVIEW

**Comment by Akshit42-hue** [comment]
cc @douglascamata @yeya24

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+	"name": "Codespaces for Thanos",
+
+	"context": "..",
+
+   	 // Using the default image so no need to specify dockerFile
+    	"dockerFile": "",
```
Comment (douglascamata) [inline .devcontainer/devcontainer.json:7]: Identation here seems off.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
```
Comment (douglascamata) [inline .devcontainer/devcontainer.json:9]: Just one empty line instead of two to separate logical keys is enough.

**Review by douglascamata — CHANGES_REQUESTED** [review]

**Comment by douglascamata** [comment]
I tried to create a Codespace from your branch and it failed:

```
2022-08-30 13:56:53.713Z: Configuration starting...
2022-08-30 13:56:53.788Z: $ git -C "/var/lib/docker/codespacemount/workspace" clone --branch main --depth 1 https://github.com/Akshit42-hue/thanos "/var/lib/docker/codespacemount/workspace/thanos"
2022-08-30 13:56:53.843Z: Cloning into '/var/lib/docker/codespacemount/workspace/thanos'...
2022-08-30 13:56:58.867Z: git process exited with exit code 0
2022-08-30 13:56:58.899Z: $ git -C "/var/lib/docker/codespacemount/workspace/thanos" config --local remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
2022-08-30 13:56:58.917Z: git process exited with exit code 0
2022-08-30 13:57:00.069Z: @microsoft/vscode-dev-containers-cli 0.66.0.
2022-08-30 13:57:00.089Z: Start: Resolving Remote
2022-08-30 13:57:00.103Z: $ docker ps -q -a --filter label=Type=codespaces
2022-08-30 13:57:00.120Z: Stop (36 ms): Run: docker ps -q -a --filter label=Type=codespaces
2022-08-30 13:57:00.133Z: $ /usr/bin/node /usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --workspace-folder /var/lib/docker/codespacemount/workspace/thanos --id-label Type=codespaces --log-level info --log-format json --config /var/lib/docker/codespacemount/workspace/thanos/.devcontainer/devcontainer.json --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --skip-post-create --update-remote-user-uid-default never --mount-workspace-git-root false
2022-08-30 13:57:00.210Z: @microsoft/vscode-dev-containers-cli 0.66.0.
2022-08-30 13:57:00.226Z: $ docker buildx version
2022-08-30 13:57:00.376Z: Stop (185 ms): Run: docker buildx version
2022-08-30 13:57:00.392Z: Start: Resolving Remote
2022-08-30 13:57:00.414Z: $ docker ps -q -a --filter label=Type=codespaces
2022-08-30 13:57:00.433Z: Stop (36 ms): Run: docker ps -q -a --filter label=Type=codespaces
2022-08-30 13:57:00.447Z: Error: EISDIR: illegal operation on a directory, read
2022-08-30 13:57:00.460Z: Stop (342 ms): Run: /usr/bin/node /usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --workspace-folder /var/lib/docker/codespacemount/workspace/thanos --id-label Type=codespaces --log-level info --log-format json --config /var/lib/docker/codespacemount/workspace/thanos/.devcontainer/devcontainer.json --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --skip-post-create --update-remote-user-uid-default never --mount-workspace-git-root false
2022-08-30 13:57:00.473Z: Exit code 1
2022-08-30 13:57:00.486Z: Failed to create container.
2022-08-30 13:57:00.506Z: Error: Command failed: /usr/bin/node /usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --workspace-folder /var/lib/docker/codespacemount/workspace/thanos --id-label Type=codespaces --log-level info --log-format json --config /var/lib/docker/codespacemount/workspace/thanos/.devcontainer/devcontainer.json --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --skip-post-create --update-remote-user-uid-default never --mount-workspace-git-root false
2022-08-30 13:57:00.521Z: Error Code: 1302
2022-08-30 13:57:00.538Z: Container creation failed.
2022-08-30 13:57:00.559Z: 
2022-08-30 13:57:00.590Z: Creating recovery container.
```

I think you don't need to provide neither the `context` nor `dockerfile` keys in the devcontainer configuration.

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	[
+		"ms-azuretools.vscode-docker",
+		"ms-kubernetes-tools.vscode-kubernetes-tools",
+		"davidanson.vscode-markdownlint",
+		"timonwong.shellcheck",
+		"docsmsft.docs-yaml",
+		"Grafana.vscode-jsonnet",
+		"codezombiech.gitignore",
+		"golang.Go"
+
```
Comment (douglascamata) [inline .devcontainer/devcontainer.json:33]: Why this blank line?

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	"context": "..",
+
+   	 // Using the default image so no need to specify dockerFile
+    	"dockerFile": "",
+
+	
+	"customizations": 
+	{
+		"vscode": {
```
Comment (douglascamata) [inline .devcontainer/devcontainer.json:12]: Inconsistent start of object here (on new line vs on same line).

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+				"settings": { 
+						"go.toolsManagement.checkForUpdates": "local",
+						"go.useLanguageServer": true,
+						"go.gopath": "/go"
```
Comment (douglascamata) [inline .devcontainer/devcontainer.json:16]: Missing indentation for keys inside object.

**Review by douglascamata — CHANGES_REQUESTED** [review]
Please do use a json formatter to format the configuration file.

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	  "ms-azuretools.vscode-docker",
+	  "ms-kubernetes-tools.vscode-kubernetes-tools",
+	  "davidanson.vscode-markdownlint",
+	  "timonwong.shellcheck",
+	  "docsmsft.docs-yaml",
+	  "Grafana.vscode-jsonnet",
+	  "codezombiech.gitignore",
+	  "golang.Go"
+	]
+}
```
Comment (yeya24) [inline .devcontainer/devcontainer.json:22]: Need an empty line

**Review by douglascamata — APPROVED** [review]
LGTM! 

I tried to create a Codespace from your fork, @Akshit42-hue, and it worked well. Thanks!

**Comment by yeya24** [comment]
Can you mark this as ready for review?

**Review by yeya24 — APPROVED** [review]
Thanks!

**Pull request merged**